nav2-usability/scenario-code/sign-in-routing/sign_in_routing_vrouter.dart

```diff
@@ -0,0 +1,197 @@
+import 'package:flutter/material.dart';
+import 'package:vrouter/vrouter.dart';
+
+void main() {
+  runApp(BooksApp());
+}
+
+class Credentials {
+  final String username;
+  final String password;
+
+  Credentials(this.username, this.password);
+}
+
+abstract class Authentication {
+  Future<bool> isSignedIn();
+
+  Future<void> signOut();
+
+  Future<bool> signIn(String username, String password);
+}
+
+class MockAuthentication implements Authentication {
+  bool _signedIn = false;
+
+  @override
+  Future<bool> isSignedIn() async {
+    return _signedIn;
+  }
+
+  @override
+  Future<void> signOut() async {
+    _signedIn = false;
+  }
+
+  @override
+  Future<bool> signIn(String username, String password) async {
+    return _signedIn = true;
+  }
+}
+
+class BooksApp extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _BooksAppState();
+}
+
+class AppState {
+  final vRouterKey = GlobalKey<VRouterState>();
+
+  final Authentication auth;
+
+  AppState(this.auth);
+
+  Future<bool> signIn(String username, String password) async {
+    var success = await auth.signIn(username, password);
+    vRouterKey.currentState!.push('/');
+    return success;
+  }
+
+  Future<void> signOut() async {
+    await auth.signOut();
+    vRouterKey.currentState!.push('/signIn');
+  }
+}
+
+class _BooksAppState extends State<BooksApp> {
+  final AppState _appState = AppState(MockAuthentication());
+
+  @override
+  Widget build(BuildContext context) {
+    return VRouter(
+      key: _appState.vRouterKey,
+      routes: [
+        VGuard(
+          beforeEnter: (vRedirector) async =>
+          (await _appState.auth.isSignedIn()) ? vRedirector.push('/') : null,
+          stackedRoutes: [
+            VWidget(
+              path: '/signIn',
+              widget: SignInScreen(
+                onSignedIn: (Credentials credentials) =>
+                    _appState.signIn(credentials.username, credentials.password),
```

Comment:
I think it makes sense to `vrouterKey` from the `AppState` class to this `State` class. AppState represents a class written by the user, and they might not want to import the `widgets` library (e.g. unit tests)

```suggestion
                onSignedIn: (Credentials credentials) async {
                  await _appState.signIn(credentials.username, credentials.password);
                  vRouterKey.currentState!.push('/');
                },

```